modules/lib/test/RequestImplementation.ts

```diff
@@ -0,0 +1,133 @@
+import type {
+    Request,
+    RequestConstructorParams,
+    RequestCookies,
+    RequestHeaders,
+} from '../core/index';
+
+export class RequestImplementation implements Request {
+    cookies: RequestCookies;
+    headers: RequestHeaders;
+    host: string;
+    method: string;
+    mode: string;
+    params: Record<string, string | string[]>;
+    path: string;
+    port: number;
+    rawPath: string;
+    remoteAddress: string;
+    scheme: string;
+    url: string;
+    webSocket: boolean;
+
+    branch?: string;
+    contextPath?: string;
+    contentType?: string;
+    repositoryId?: string;
+    validTicket?: boolean;
+    body?: string;
+
+    constructor({
+        // Required
+        cookies,
+        headers,
+        host,
+        method,
+        mode,
+        params,
+        path,
+        port,
+        rawPath,
+        remoteAddress,
+        scheme,
+        url,
+        webSocket,
+        // Optionals
+        branch,
+        contextPath,
+        contentType,
+        repositoryId,
+        validTicket,
+        // Special
+        body,
+    }: RequestConstructorParams) {
+        this.cookies = cookies;
+        this.headers = headers;
+        this.host = host;
+        this.method = method;
+        this.mode = mode;
+        this.params = params;
+        this.path = path;
+        this.port = port;
+        this.rawPath = rawPath;
+        this.remoteAddress = remoteAddress;
+        this.scheme = scheme;
+        this.url = url;
+        this.webSocket = webSocket;
+
+        this.branch = branch;
+        this.contextPath = contextPath;
+        this.contentType = contentType;
+        this.repositoryId = repositoryId;
+        this.validTicket = validTicket;
+
+        this.body = body;
+    }
+
+    getHeader(header: string): string | null {
+        const value = this.headers[header];
```

Comment:
actual implementation is case-insensitive

Reply:
RIght, that file is just a test, but good to know. That class could be useful in mock-xp.